Installation of persistence-jdbc-postgresql-1.9.0.b4 via PaperUI fails

As suggested by kaikreuzer this issue was moved here from https://github.com/openhab/openhab2-addons/issues/1293

Test setup: Genuine freshly unzipped v2.0.0.b4 #477 (offline). Only change in conf/services/jdbc.cfg: Adding of the database parameters:

`url=jdbc:postgresql://localhost:5432/hab2`
`user=hab`
`password=xxx`

Installing persistence-jdbc-postgresql-1.9.0.b4 via PaperUI > Extensions > Persistence > JDBC Persistence PostgreSQL > Install

The installation does not complete (The wait icon remains spinning) and the log states the following exception:

`[ERROR] [st.core.extensions.ExtensionResource] - Exception while installing extension: org.apache.karaf.features.internal.util.MultiException: Error restarting bundles`

Afterwards when restarting openhab the following exception appears in the log:

`[ERROR] [org.postgresql.jdbc41               ] - FrameworkEvent ERROR - org.postgresql.jdbc41
org.osgi.framework.BundleException: Exception in org.postgresql.osgi.PGBundleActivator.start() of bundle org.postgresql.jdbc41.
        at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:792)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:721)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:936)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:319)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.Module.doStart(Module.java:571)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.Module.start(Module.java:439)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1582)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1562)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1533)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1476)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
Caused by: java.lang.NoClassDefFoundError: org/osgi/service/jdbc/DataSourceFactory
        at org.postgresql.osgi.PGBundleActivator.start(PGBundleActivator.java:32)
        at org.eclipse.osgi.internal.framework.BundleContextImpl$3.run(BundleContextImpl.java:771)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.internal.framework.BundleContextImpl$3.run(BundleContextImpl.java:1)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at java.security.AccessController.doPrivileged(Native Method)[:1.8.0_91]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:764)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        ... 12 more
Caused by: java.lang.ClassNotFoundException: org.osgi.service.jdbc.DataSourceFactory cannot be found by org.postgresql.jdbc41_9.4.0.build-1206
        at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:432)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:345)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:337)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:160)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)[:1.8.0_91]
        ... 17 more
`

Kind regards,
brstein.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/38194733-installation-of-persistence-jdbc-postgresql-1-9-0-b4-via-paperui-fails?utm_campaign=plugin&utm_content=tracker%2F482064&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F482064&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/internal/JdbcConfiguration.java
+++ b/bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/internal/JdbcConfiguration.java
@@ -291,7 +291,7 @@ public class JdbcConfiguration {
             } else if (serviceName.equals("mysql")) {
                 warn += "\tMySQL:     version >= 5.1.36 from             http://mvnrepository.com/artifact/mysql/mysql-connector-java\n";
             } else if (serviceName.equals("postgresql")) {
-                warn += "\tPostgreSQL:version >= 9.4-1201-jdbc41 from    http://mvnrepository.com/artifact/org.postgresql/postgresql\n";
+                warn += "\tPostgreSQL:version >= 9.4.1208 from    http://mvnrepository.com/artifact/org.postgresql/postgresql\n";
             } else if (serviceName.equals("sqlite")) {
                 warn += "\tSQLite:    version >= 3.8.11.2 from           http://mvnrepository.com/artifact/org.xerial/sqlite-jdbc\n";
             }

--- a/pom.xml
+++ b/pom.xml
@@ -56,16 +56,16 @@
         <maven.compiler.target>1.7</maven.compiler.target>
 
         <repo.url>https://api.bintray.com/maven/openhab/mvn/openhab</repo.url>
-		 
+
 		<!-- JDBC Databasedriver Versions -->
 		<derby.version>10.12.1.1</derby.version>
 		<h2.version>1.4.191</h2.version>
 		<hsqldb.version>2.3.3</hsqldb.version>
 		<mariadb.version>1.3.5</mariadb.version>
 		<mysql.version>5.1.38</mysql.version>
-		<postgresql.version>9.4-1206-jdbc41</postgresql.version>
+		<postgresql.version>9.4.1211</postgresql.version>
 		<sqlite.version>3.8.11.2</sqlite.version>
-		
+
     </properties>
 
     <packaging>pom</packaging>

